core/shard_thread.go

```diff
@@ -0,0 +1,129 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/dicedb/dice/core/auth"
+	"github.com/dicedb/dice/core/ops"
+
+	"github.com/dicedb/dice/config"
+	"github.com/dicedb/dice/core/diceerrors"
+	"github.com/dicedb/dice/server/utils"
+)
+
+type ShardID int8
+
+type ShardError struct {
+	shardID ShardID // shardID is the ID of the shard that encountered the error
+	err     error   // err is the error that occurred
+}
+
+type ShardThread struct {
+	id               ShardID                            // id is the unique identifier for the shard.
+	store            *Store                             // store that the shard is responsible for.
+	ReqChan          chan *ops.StoreOp                  // ReqChan is this shard's channel for receiving requests.
+	workerMap        map[string]chan *ops.StoreResponse // workerMap maps workerID to its unique response channel
```

Comment:
nit: Would it make sense to move this to `sync.Map`? This might help us remove explicit locking.
If worker register/unregister are infrequent we might consider this.

Reply:
sync map is useful in scenarios where there are high number of reads and fewer writes, in our case we have both happening at a relatively high number.
Based on the documentation for sync.Map, it seems like using a mutex is better suited here: https://pkg.go.dev/sync#Map

Let me know if you think I am missing something.

Reply:
We can check Swiss maps that Arpit introduced. It appears to be concurrent out of the box.

Reply:
I would suggest taking this as part of a separate PR and migrating all our maps to Swiss Map. There is already too much happening in this PR...

Reply:
Sure